modelling/src/main/java/org/axonframework/modelling/HierarchicalStateManagerConfigurationEnhancer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.configuration.ComponentRegistry;
+import org.axonframework.configuration.ConfigurationEnhancer;
+
+import java.util.Optional;
+
+/**
+ * {@link ConfigurationEnhancer} that registers a decorator for the {@link StateManager} that, when a parent
+ * configuration is present, wraps child and parent {@link StateManager} in a {@link HierarchicalStateManager} that
+ * delegates to the parent if a state cannot be resolved by the current configuration.
+ * <p>
+ * To prevent modules not defining entities not being able to find the {@link StateManager} component to access the
```

Comment:
Given all the changes Allard's put on this stuff, I am not sure whether the following is still the case. But, I have to ask. If there is no registered component, will the decorator you register here even be invoked? If not, this sentence is (for now) incorrect. However, I do think it's wise to register the given `component` on the decorator lambda as is if there is no parent `StateManager` for sure.

Reply:
I deleted a piece of code here that I am going to readd:

```java
if (!componentRegistry.hasComponent(StateManager.class)) {
    componentRegistry.registerComponent(
            StateManager.class,
            (c) -> SimpleStateManager.builder("Empty").build()
    );
}
```

I am not 100% it's needed, that's why I removed it. But this way the javadoc is in line again, and I don't think it can hurt.

Basically, it makes an empty StateManager to prevent null problems and hierarchy issues.

Reply:
Sounds like a fair addition to me!